libs/watt/package/text-field/watt-text-field.component.ts

```diff
@@ -112,45 +116,54 @@ export type WattInputTypes = 'text' | 'password' | 'email' | 'number' | 'tel' |
   </watt-field>`,
 })
 export class WattTextFieldComponent implements ControlValueAccessor, AfterViewInit {
-  @Input() value!: string;
-  @Input() type: WattInputTypes = 'text';
-  @Input() placeholder?: string;
-  @Input() label = '';
-  @Input() tooltip?: string;
-  @Input() prefix?: WattIcon;
-  @Input() maxLength: string | number | null = null;
-  @Input() formControl!: FormControl;
-  @Input() autocompleteOptions!: string[];
-  @Input() autocompleteMatcherFn!: (value: string, option: string) => boolean;
+  value = input<string>('');
+  type = input<WattInputTypes>('text');
+  placeholder = input<string | undefined>();
+  label = input<string>('');
+  tooltip = input<string | undefined>();
+  prefix = input<WattIcon | undefined>();
+  maxLength = input<string | number | null>();
+  formControl = input.required<FormControl>();
+  autocompleteOptions = input<string[] | undefined>([]);
+  autocompleteMatcherFn = input<((value: string, option: string) => boolean) | undefined>();
 
   /** @ignore */
-  @ViewChild(MatAutocomplete) autocompleteRef!: MatAutocomplete;
+  autocompleteRef = viewChild(MatAutocomplete);
 
   /**
    * Emits the value of the input field when it changes.
    */
-  @Output() searchChanged = new EventEmitter<string>();
+  searchChanged = output<string>();
 
   /**
    * Emits the value of the input field when an autocomplete option is selected.
    */
-  @Output() autocompleteOptionSelected = new EventEmitter<string>();
+  autocompleteOptionSelected = output<string>();
 
   /**
    * Emits the value of the input field when an autocomplete option is selected.
    */
-  @Output() autocompleteOptionDeselected = new EventEmitter<void>();
+  autocompleteOptionDeselected = output<void>();
 
   /** @ignore */
   private element = inject(ElementRef);
 
   /** @ignore */
-  @ViewChild('inputField') inputField!: ElementRef<HTMLInputElement>;
-  model!: string;
+  inputField = viewChild<ElementRef<HTMLInputElement>>('inputField');
+  model = model<string>('');
+
+  constructor() {
+    // Sync input value with model
+    effect(() => {
+      const inputValue = this.value();
+      if (inputValue !== undefined) {
+        this.model.set(inputValue);
+      }
+    });
+  }
```

Comment:
question: should `model` actually be exposed as an input, as in do you need to be able to set it in the template? Like this:

```html
<watt-text-field [model]="someValue()" />
```

If not, then I think `linkedSignal` is much better here - you could also get rid of the `effect` in that case:

```ts
// you can call `.set` on the model signal to update it and if the `value` input changes,
// then model will automatically be synchronized to reflect the new input.
model = linkedSignal(this.value);
```